consonance-webservice/src/main/java/io/swagger/wes/api/impl/Ga4ghApiServiceImpl.java

```
@@ -0,0 +1,466 @@
+                });
+
+        LOG.info(workflowStatus.toString());
+
+        return Response.ok().entity(workflowStatus).build();
+    }
+
+    /***
+     * If there is jobs in the queue, and running.
+     * Listing workflows will show their Id, and Status.
+     */
+    @Override
+    public Response listWorkflows(Long pageSize, String pageToken, String keyValueSearch, ConsonanceUser user) throws NotFoundException {
+        Long optionalPageSize = pageSize;
+        String optionalPageToken = pageToken;
+        String optionalKeyFilter = keyValueSearch;
+
+        Ga4ghWesWorkflowListResponse list = new Ga4ghWesWorkflowListResponse();
+
+        List<Job> allJobs = orderResource.listWorkflowRuns(user);
+        Map <Integer, JobState> serialize = new TreeMap<Integer, JobState>();
+
+        allJobs.stream().forEach((Job j) -> serialize.put(j.getJobId(), j.getState()));
+
+        for (Integer key: serialize.keySet()){
+            Ga4ghWesWorkflowDesc descriptor = new Ga4ghWesWorkflowDesc();
+            descriptor.setWorkflowId(String.valueOf(key));
+            descriptor.setState(mapState(serialize.get(key)));
+            list.addWorkflowsItem(descriptor);
+        }
+
+        return Response.ok().entity(list).build();
+    }
+
+    /**
+     * Calls on existing method to schedule a job.
+     * Requires user with the request to be authenticated
+     */
+    @Override
+    public Response runWorkflow(Ga4ghWesWorkflowRequest body, ConsonanceUser user) throws NotFoundException, IOException, TimeoutException {
+
+        Map workflowKeyValues = body.getKeyValues();
+
+        // TODO: Check if version in body, matches consonance version. Confirm non-empty value.
+        String workflowTypeVersion = body.getWorkflowTypeVersion(); // Version of runner tool.
+        if (workflowTypeVersion.isEmpty()) {
+            return Response.noContent().build();
+        } else {
+            String currentRunnerVersion = "RandomVersionTEMP";
```

Should you throw an error in this case instead?

---

Removing `else` statement.